scripts/vm_run_tests.psm1

```diff
@@ -636,84 +380,89 @@ function Invoke-ConnectRedirectTestsOnVM
-        param([Parameter(Mandatory = $True)] [string] $WorkingDirectory,
-              [Parameter(Mandatory = $True)] [string] $LogFileName)
-        $WorkingDirectory = "$Env:SystemDrive\$WorkingDirectory"
+function Stop-eBPFComponents {
+    param(
+        [Parameter(Mandatory = $false)] [bool] $VerboseLogs = $false,
+        [Parameter(Mandatory = $true)] [string] $LogFileName
+    )
+    Write-Log "Stopping eBPF components"
+    $scriptBlock = {
+        param($WorkingDirectory, $LogFileName)
         Import-Module $WorkingDirectory\common.psm1 -ArgumentList ($LogFileName) -Force -WarningAction SilentlyContinue
         Import-Module $WorkingDirectory\install_ebpf.psm1 -ArgumentList ($WorkingDirectory, $LogFileName) -Force -WarningAction SilentlyContinue
-
         Stop-eBPFComponents
-    } -ArgumentList ("eBPF", $LogFileName) -ErrorAction Stop
+    }
+    $argList = @($script:WorkingDirectory, $LogFileName)
+    Invoke-OnHostOrVM -ScriptBlock $scriptBlock -ArgumentList $argList
 }
 
-function Run-KernelTestsOnVM
-{
-    param([Parameter(Mandatory = $true)] [string] $VMName,
-          [Parameter(Mandatory = $true)] [PSCustomObject] $Config)
-
-    # Run CICD tests on test VM.
-    Invoke-CICDTestsOnVM `
-        -VMName $VMName `
-        -TestMode $TestMode `
-        -Options $Options
-
-    # The required behavior is selected by the $TestMode
-    # parameter.
-    if (($TestMode -eq "CI/CD") -or ($TestMode -eq "Regression")) {
-
-        # Run XDP Tests.
-        Invoke-XDPTestsOnVM `
-            -Interfaces $Config.Interfaces `
-            -VMName $VMName
-
-        # Run Connect Redirect Tests.
-        Invoke-ConnectRedirectTestsOnVM `
-            -Interfaces $Config.Interfaces `
-            -ConnectRedirectTestConfig $Config.ConnectRedirectTest `
-            -UserType "Administrator" `
-            -VMName $VMName
-
-        Invoke-ConnectRedirectTestsOnVM `
-            -Interfaces $Config.Interfaces `
-            -ConnectRedirectTestConfig $Config.ConnectRedirectTest `
-            -UserType "StandardUser" `
-            -VMName $VMName
+function Run-KernelTests {
+    param(
+        [Parameter(Mandatory = $true)] [PSCustomObject] $Config,
+        [Parameter(Mandatory = $false)] [bool] $VerboseLogs = $false
+    )
+    Write-Log "Execute Run-KernelTests"
+    $scriptBlock = {
+        param($WorkingDirectory, $VerboseLogs, $TestMode, $TestHangTimeout, $UserModeDumpFolder, $Options, $LogFileName)
+        Import-Module $WorkingDirectory\common.psm1 -ArgumentList ($LogFileName) -Force -WarningAction SilentlyContinue
+        Import-Module $WorkingDirectory\run_driver_tests.psm1 -ArgumentList ($WorkingDirectory, $LogFileName, $TestHangTimeout, $UserModeDumpFolder) -Force -WarningAction SilentlyContinue
+        $TestMode = $TestMode.ToLower()
+        switch ($TestMode) {
+            "ci/cd" {
+                # TODO - possibly these write-logs are not valid?
```

Comment:
Another TODO.  At least add a github issue number.

Reply:
Thanks for pointing this out - this was intended for cleanup before check-in. I have removed this.